### ISSUE TITLE
Missing Docker Images causes tests to fail

### ISSUE THREAD

**Issue by BenHall** [ContainerSolutions/minimesos#250]
When running the build on a new machine, the tests fail because of missing Docker Images. Ideally, this should download the images if missing.

```
com.containersolutions.mesoshelloworld.systemtest.DiscoverySystemTest > classMethod FAILED
    com.containersol.minimesos.MinimesosException: Failed to start hello-world-scheduler_1 (null) container
        at com.containersol.minimesos.cluster.MesosCluster.addAndStartContainer(MesosCluster.java:272)
        at com.containersol.minimesos.cluster.MesosCluster.addAndStartContainer(MesosCluster.java:286)
        at com.containersolutions.mesoshelloworld.systemtest.DiscoverySystemTest.startScheduler(DiscoverySystemTest.java:48)

        Caused by:
        com.github.dockerjava.api.NotFoundException: No such image: containersol/mesos-hello-world-scheduler:latest
            at com.github.dockerjava.jaxrs.filter.ResponseStatusExceptionFilter.filter(ResponseStatusExceptionFilter.java:47)
            at org.glassfish.jersey.client.ClientFilteringStages$ResponseFilterStage.apply(ClientFilteringStages.java:134)
            at org.glassfish.jersey.client.ClientFilteringStages$ResponseFilterStage.apply(ClientFilteringStages.java:123)
            at org.glassfish.jersey.process.internal.Stages.process(Stages.java:171)
            at org.glassfish.jersey.client.ClientRuntime.invoke(ClientRuntime.java:251)
            at org.glassfish.jersey.client.JerseyInvocation$2.call(JerseyInvocation.java:683)
            at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
            at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
            at org.glassfish.jersey.internal.Errors.process(Errors.java:228)
            at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:424)
            at org.glassfish.jersey.client.JerseyInvocation.invoke(JerseyInvocation.java:679)
            at org.glassfish.jersey.client.JerseyInvocation$Builder.method(JerseyInvocation.java:435)
            at org.glassfish.jersey.client.JerseyInvocation$Builder.post(JerseyInvocation.java:338)
            at com.github.dockerjava.jaxrs.CreateContainerCmdExec.execute(CreateContainerCmdExec.java:33)
            at com.github.dockerjava.jaxrs.CreateContainerCmdExec.execute(CreateContainerCmdExec.java:14)
            at com.github.dockerjava.jaxrs.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:24)
            at com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:33)
            at com.github.dockerjava.core.command.CreateContainerCmdImpl.exec(CreateContainerCmdImpl.java:142)
            at com.containersol.minimesos.container.AbstractContainer.start(AbstractContainer.java:76)
            at com.containersol.minimesos.cluster.MesosCluster.addAndStartContainer(MesosCluster.java:268)
            ... 2 more
```

**Comment by sadovnikov** [comment]
@BenHall something is wrong with your environment or, most probably, we did not describe it well enough. These images do get built as a part of the build... before being used, of course ;)

**Comment by sadovnikov** [comment]
Found it https://github.com/ContainerSolutions/minimesos/pull/177#issuecomment-164394946

Images are built as part of Gradle `install`, but used in `test`.
We've got to build them in `test`

**Comment by BenHall** [comment]
@sadovnikov Yes, sorry, not a great example. As you pointed out, this failed because the overall build failed. I was trying to speed up execution so just running the tests. When I downloaded all the main images this was the next one missing. After a complete build everything passed successfully. 

The main problem is with the other images missing which are already on the hub. The only fix was to download before running the tests. I've added `make deps` as a quick fix until this is closed. 

I've also experienced this when running `minimesos up`

Error pulling image or image not found in registry: mesosphere/marathon:v0.13.0                         
Error during callback                                                                                   
java.lang.RuntimeException: Error pulling image or image not found in registry: mesosphere/marathon:v0.1
3.0                                                                                                     
        at com.containersol.minimesos.container.AbstractContainer$2.onNext(AbstractContainer.java:191) ~
[minimesos.jar:0.6.0]                                                                                   
        at com.containersol.minimesos.container.AbstractContainer$2.onNext(AbstractContainer.java:181) ~
[minimesos.jar:0.6.0]                                                                                   
        at com.github.dockerjava.jaxrs.AbstrAsyncDockerCmdExec$1.onNext(AbstrAsyncDockerCmdExec.java:42)
 ~[minimesos.jar:0.6.0]                                                                                 
        at com.github.dockerjava.core.async.JsonStreamProcessor.processResponseStream(JsonStreamProcesso
r.java:45) ~[minimesos.jar:0.6.0]                                                                       
        at com.github.dockerjava.jaxrs.async.AbstractCallbackNotifier.call(AbstractCallbackNotifier.java
:62) [minimesos.jar:0.6.0]                                                                              
        at com.github.dockerjava.jaxrs.async.AbstractCallbackNotifier.call(AbstractCallbackNotifier.java
:22) [minimesos.jar:0.6.0]                                                                              
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_66-internal]              
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_66-
internal]                                                                                               
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_66-
internal]                                                                                               
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_66-internal]
